examples/pytorch/gxn/README.md

```diff
@@ -0,0 +1,108 @@
+- NumFeats: -
+- NumClasses: 3
+
+
+How to run example files
+--------------------------------
+If you want to reproduce the author's result, in the GXN-DGL folder, run
+
+```bash
+bash scripts/run_gxn.sh ${dataset_name} ${device_id} ${num_trials} ${print_trainlog_every}
+```
+
+If you want to perform a early-stop version experiment, in the GXN-DGL folder, run
+
+```bash
+bash scripts/run_gxn_early_stop.sh ${dataset_name} ${device_id} ${num_trials} ${print_trainlog_every}
+```
+
+where
+- dataset_name: Dataset name used in this experiment. Could be DD', 'PROTEINS', 'ENZYMES', 'IMDB-BINARY', 'IMDB-MULTI' and 'COLLAB'.
+- device_id: ID of computation device. -1 for pure CPU computation. For example if you only have single GPU, set this value to be 0.
+- num_trials: How many times does the experiment conducted.
+- print_training_log_every: Print training log every ? epochs. -1 for silent training.
+
+
+NOTE: If your have problem when using 'IMDB-BINARY', 'IMDB-MULTI' and 'COLLAB', it could be caused by a bug in `LegacyTUDataset`/`TUDataset` in DGL (see [here](https://github.com/dmlc/dgl/pull/2543)). Try update your DGL or
+- use `TUDataset` with `force_reload=True`
+- delete dataset files 
+- change `degree_as_feature(dataset)` and `node_label_as_feature(dataset, mode=mode)` to `degree_as_feature(dataset, save=False)` and `node_label_as_feature(dataset, mode=mode, save=False)` in `main.py`.
+
+Performance
+-------------------------
+
+**Accuracy**
+
+**NOTE**: Different from our implementation, the author uses fixed dataset split. Thus there may be difference between our result and the author's result. **To compare our implementation with the author's, we follow the setting in the author's implementation that performs model-selection on testset**. We also try early-stop with patience equals to 1/5 of the total number of epochs for some datasets. The result of `Author's Code` in the table below are obtained using first-ford data as the test dataset.
+
+|                   | DD           | PROTEINS    | ENZYMES     | IMDB-BINARY | IMDB-MULTI | COLLAB     |
+| ------------------| ------------ | ----------- | ----------- | ----------- | ---------- | ---------- |
+| Reported in Paper | 82.68(4.1 )  | 79.91(4.1)  | 57.50(6.1)  | 78.60(2.3)  | 55.20(2.5) | 78.82(1.4) |
+| Author's Code     | 82.05        | 72.07       | 58.33       | 77.00       | 56.00      | 80.40      |
+| DGL               | 82.97(3.0)   | 78.21(2.0)  | 57.50(5.5)  | 78.70(4.0)  | 52.26(2.0) | 80.58(2.4) |
+| DGL(early-stop)   | 78.66(4.3)   | 73.12(3.1)  | 39.83(7.4)  | N/A         | N/A        | N/A        |
+
+
+**Speed**
+
+Device: 
+- CPU: Intel(R) Xeon(R) CPU E5-2686 v4 @ 2.30GHz
+- GPU: Tesla V100-SXM2 16GB
+
+In seconds
+
+|               | DD    | PROTEINS | ENZYMES | IMDB-BINARY | IMDB-MULTI | COLLAB |
+| ------------- | ----- | -------- | ------- | ----------- | ---------- | ------ |
+| Author's Code | 25.32 | 2.93     | 1.53    | 2.42        | 3.58       | 19.78  |
```

Comment:
Is there explanation on DGL being slower on COLLAB?

Reply:
I may need to check the author's code. As you can see, the training time on DD, which is a dataset with fewer edges (~716/graph) than COLLAB (~2500/graph), is longer than the time on COLLAB.

Reply:
It seems that CPU is faster on small batch size.